src/value/owned.rs

```diff
@@ -233,12 +244,7 @@ impl Struct for OwnedStruct {
 impl PartialEq for OwnedStruct {
     fn eq(&self, other: &Self) -> bool {
         for (key, value) in &self.text_fields {
-            let other_values: Vec<&OwnedElement> = other.get_all(key).collect();
-            if other_values.is_empty() {
-                return false;
-            }
-            let this_values: Vec<&OwnedElement> = value.iter().map(|(_s, v)| v).collect();
-            if other_values != this_values {
+            if other.get_all(key).ne(value.iter().map(|(_s, v)| v)) {
                 return false;
             }
```

Comment:
As I look at this I think some separate struct tests are in order, especially because I think we need to take `no_text_fields` into account.

Specifically, I wouldn't try to roll all of this testing into the value tests, I think they are complicated to have their own sets.

Reply:
Also, what happens if the vector order is different for a given key, this technically doesn't work, does it?

In other words `{a: 1, a: 2}` should equal `{a: 2, a: 1}`, does that work?

Reply:
I misread the code above--it should work (though tests needed), but it is an O(N**2) comparison, If we want to make this O(N lg N) or O(N), we need an intermediate `BTreeSet` or `HashSet`.

Reply:
See #160.